bundle/deploy/files/delete.go

```diff
@@ -46,20 +48,36 @@ func (m *delete) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
 	}
 
 	// Clean up sync snapshot file
-	sync, err := GetSync(ctx, b)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	err = sync.DestroySnapshot(ctx)
+	err = deleteSnapshotFile(ctx, b)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	cmdio.LogString(ctx, fmt.Sprintf("Deleted snapshot file at %s", sync.SnapshotPath()))
 	cmdio.LogString(ctx, "Successfully deleted files!")
 	return nil
 }
 
+func deleteSnapshotFile(ctx context.Context, b *bundle.Bundle) error {
+	cacheDir, err := b.CacheDir(ctx)
+	if err != nil {
+		return err
+	}
+	sp, err := sync.SnapshotPath(&sync.SyncOptions{
+		SnapshotBasePath: cacheDir,
+		Host:             b.WorkspaceClient().Config.Host,
+		RemotePath:       b.Config.Workspace.FilePath,
+	})
```

Comment:
Good point. Reusing `GetSyncOptions`. Reusing `GetSync` is tricky because it has an `EnsureRemotePathIsUsable` call that ensures the sync destination exists (by creating it if needed, the root cause of the bug this solves).